docs/latest/design/bootstrap.md

```diff
@@ -0,0 +1,374 @@
+# Bootstrap Design
+
+## Overview
+
+[Issue 31][] specifies the need for allowing advanced users to specify their custom
+Envoy Bootstrap configuration rather than using the default Bootstrap configuration
+defined in Envoy Gateway. This allows advanced users to extend Envoy Gateway and
+support their custom use cases such setting up tracing and stats configuration
+that is not supported by Envoy Gateway.
+
+## Goals
+
+* Define an API field to allow a user to specify a custom Bootstrap
+* Provide tooling to allow the user to generate the default Bootstrap configuration
+  as well as validate their custom Bootstrap.
+
+## Non Goals
+
+* Allow user to configure only a section of the Bootstrap
+
+## API
+
+Leverage the existing [EnvoyProxy][] resource which can be attached to the [GatewayClass][] using
```

Comment:
I wonder if EnvoyProxy Config can be attached to `Gateway`, I think maybe `GatewayClass` is effected all Gateways, right? Can end user just override single Gateway?

Reply:
Make sences, thanks @arkodg